### PR TITLE
Add sorting to kubectl top

### PR DESCRIPTION
**Pull request by RemingtonReackhof**
Addresses part of #30507 
```release-note
**What this PR does / why we need it**: This PR  adds functionality to the kubectl top command to sort by a given resource name: name, cpu, or memory.
**Which issue this PR fixes**: Addresses part of #30507
**Special notes for your reviewer**: Due to the way container metrics were added when displaying pods beforehand, I had to refactor the way in which it was being done. I believe it is clearer now than before.
```

### CODE REVIEW

File: pkg/kubectl/metricsutil/metrics_sorter.go
Comment (soltysh) [inline pkg/kubectl/metricsutil/metrics_sorter.go]: We're in 2017 right now ;)

File: pkg/kubectl/cmd/top_node.go
Comment (soltysh) [inline pkg/kubectl/cmd/top_node.go]: Maybe use `api.Resource*` constants here.

File: pkg/kubectl/cmd/top_node.go
Comment (soltysh) [inline pkg/kubectl/cmd/top_node.go]: I don't think you need to read that again manually. You've already assinged `SortBy` member to `sort-by` flag.

File: pkg/kubectl/cmd/top_node.go
Comment (soltysh) [inline pkg/kubectl/cmd/top_node.go]: Not needed, either.

File: pkg/kubectl/metricsutil/metrics_printer.go
Comment (soltysh) [inline pkg/kubectl/metricsutil/metrics_printer.go]: What if `sorting` is empty? Same thing in the else.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by soltysh** [comment]
I've left some comments, but wrt overall design I'd defer to @piosz since he was mentoring the previous person working on it and @kubernetes/sig-cli-maintainers

**Comment by piosz** [comment]
High level looks reasonable. I'm not an owner of this component so I'll let @kubernetes/sig-cli-maintainers to decide.

**Comment by jonaz** [comment]
why was this not merged?

**Comment by dims** [comment]
looks like it timed out waiting for reviewers

**Comment by piosz** [comment]
@kubernetes/kubectl-maintainers for review

**Pull request closed without merging**
